feat: Drag a pageBlock into the left nav

mattkrick opened the pull request:
# Description

fix #12192

https://www.loom.com/share/0f49524a634b453eb23ae744e60da290


PageBlocks should be droppable all over the place
- [x] drop them on left nav items (teams and pages)
- [x] drop them below left nav items (pages, where appropriate)

We can do the following on another PR
- [ ] drop them in bread crumbs
- [ ] drop them on other page blocks? (we may need an animation for this so people know where it went)
- [ ] drop left nav pages onto a Page?
- [ ] drop left nav pages onto a breadcrumb?

Dschoordsch reviewed (dismissed):
Highlighting does not work on Safari and when moving a page link, the database attribute gets lost https://www.loom.com/share/14275f609d27413fbe8de0739c2facd7

mattkrick commented:
fixed! `relatedTarget` doesn't exist in safari 😢. The database attribute wasn't getting copied across other things either so I fixed that up everywhere i found it.
